src/pextlib1.0/Pextlib.c

```diff
@@ -648,6 +651,424 @@ int SetMaxOpenFilesCmd(ClientData clientData UNUSED, Tcl_Interp *interp, int obj
+
+            free(new_entry);
+
+            return ret;
+        }
+
+        new_entry->case_sensitive = case_sensitive;
+
+        cache->entries[cache->count++] = new_entry;
+
+        ret = 0;
+    }
+
+    return ret;
+}
+
+/**
+ * Looks up a cached mountpoint case-sensitivity value.
+ *
+ * Returns 1 if the mountpoint's FS is case-sensitive,
+ * 0 if it's case-insensitive and -1 on error or if
+ * no such entry was found.
+ */
+static int lookup_mount_cs_cache(mount_cs_cache_t *cache, const char *mountpoint) {
+    int ret = -1;
+
+    /* No specified mount point doesn't make sense, so assert it. */
+    assert(mountpoint);
+
+    if (cache && mountpoint) {
+        for (size_t i = 0; i < cache->count; ++i) {
+            if ((cache->entries[i]) && (cache->entries[i]->mountpoint)) {
+                if (0 == strcmp(cache->entries[i]->mountpoint, mountpoint)) {
+                    return cache->entries[i]->case_sensitive;
+                }
+            }
+        }
+    }
+
+    return ret;
+}
+
+
+/**
+ * Gets the corresponding mount point for a file.
+ * Returns NULL if looking up the mount point was not possible.
+ */
+static char* get_mntpoint(const char *path) {
+    char *ret = NULL;
+
+#if defined(__APPLE__) || defined(__OpenBSD__) || defined(__FreeBSD__) || defined(__NetBSD__)
+    struct statfs f = { 0 };
+
+    if (-1 != statfs(path, &f)) {
+        if (f.f_mntonname) {
+            ret = strdup(f.f_mntonname);
+        }
+    }
+#else
+    /*
+     * Systems like Solaris, IRIX, True64, AIX and others have no way to get this information easily.
+     *
+     * Neither does Linux. We could go ahead and try to "compute" the mount point
+     * using a series of stat calls and the like, but it doesn't make any sense.
+     * Running our three lstat() calls should be less resource hungry than getting
+     * the mount point on this system, so disable caching there as well.
+     */
+    ret = NULL;
+#endif
+
+    return ret;
+}
+
+#ifdef __APPLE__
+
+#include <sys/attr.h>
+
+typedef struct volcaps {
+      u_int32_t size;
+      vol_capabilities_attr_t volcaps;
+} volcaps_t;
+
+/**
+ * Default function for determining the FS case sensitivity on Darwin.
+ * Using getattrlist().
+ */
+int fs_case_sensitive_darwin(Tcl_Interp *interp, const char *path, mount_cs_cache_t *cache) {
+    int ret = -1;
+
+    if (!path) {
+        return ret;
+    }
+
+    char *mntpoint = get_mntpoint(path);
+
+    if (!mntpoint) {
+        return ret;
+    }
+
+    if (cache) {
+        ret = lookup_mount_cs_cache(cache, mntpoint);
+
+        if (-1 != ret) {
+            free(mntpoint);
+            return ret;
+        }
+    }
+
+    struct attrlist attrlist = { 0 };
+    volcaps_t volcaps = { 0 };
+
+    attrlist.bitmapcount = ATTR_BIT_MAP_COUNT;
+    attrlist.volattr = ATTR_VOL_CAPABILITIES;
+
+    if (-1 == getattrlist(mntpoint, &attrlist, &volcaps, sizeof(volcaps), 0)) {
+        free(mntpoint);
+        return ret;
+    }
+
+    if (-1 == ret) {
```

Comment:
`ret` will always be -1 here. There is no code path that could make it a different value.